Classes/Decoder/UrlDecoder.php

```diff
@@ -1225,7 +1204,7 @@ protected function searchForPathOverrideInPagesLanguageverlay($path) {
 				$result = GeneralUtility::makeInstance('DmitryDulepov\\Realurl\\Cache\\PathCacheEntry');
 				/** @var \DmitryDulepov\Realurl\Cache\PathCacheEntry $result */
 				$result->setLanguageId($this->detectedLanguageId);
-				$result->setPageId((int)$row['pid']);
+				$result->setPageId((int)$row['uid']);
```

Comment:
At this line the result pid is always 0 (zero) because of the select statement above.